test/extensions/common/aws/credentials_provider_impl_test.cc

```diff
@@ -2733,10 +2859,11 @@ class DefaultCredentialsProviderChainTest : public testing::Test {
     EXPECT_CALL(factories_, createEnvironmentCredentialsProvider());
   }
 
-  ~DefaultCredentialsProviderChainTest() override {
+  void SetUp() override {
```

Comment:
This appears to be a long standing test bug - we should be clearing these environment variables as part of setup, instead by doing in destructor means that previous environment variable set affects test case 1 (which happened in this new patch)

Reply:
SetUp is called implicitly for each fixture execution.
Have added comment to indicate.